evalml/pipelines/components/transformers/encoders/target_encoder.py

```diff
@@ -0,0 +1,67 @@
+from ..transformer import Transformer
+
+from evalml.pipelines.components import ComponentBaseMeta
+from evalml.utils import import_or_raise
+
+
+class TargetEncoderMeta(ComponentBaseMeta):
```

Comment:
nit-pick: I think we can reuse the OneHotEncoderMeta. If a user calls `categories` on TargetEncoder, they'll get an `AttributeError` instead of `ComponentNotYetFittedError` which is what I'd expect. 

It's a minor point but it would be nice if our encoders can share as much code as possible, in my opinion.

Reply:
Great catch, will fix